### ISSUE TITLE
False no-undef when destructuring object with type

### ISSUE THREAD

**Issue by ewired** [denoland/deno_lint#640]
When destructuring an object in an arrow function argument and defining its type, the linter reports a no-undef warning for the argument following the object, and reports a no-undef warning for all other references to the argument.
```ts
export const myFunction = (
  { thing }: { thing: string[] },
  somethingElse = false // somethingElse is not defined
) => {
  return somethingElse ? thing : ["stuff"]; // somethingElse is not defined
};

{
	"resource": "/c:/project/thing.ts",
	"owner": "deno",
	"code": "no-undef",
	"severity": 4,
	"message": "somethingElse is not defined",
	"source": "deno-lint",
	"startLineNumber": 3,
	"startColumn": 3,
	"endLineNumber": 3,
	"endColumn": 16
}
{
	"resource": "/c:/project/thing.ts",
	"owner": "deno",
	"code": "no-undef",
	"severity": 4,
	"message": "somethingElse is not defined",
	"source": "deno-lint",
	"startLineNumber": 5,
	"startColumn": 10,
	"endLineNumber": 5,
	"endColumn": 23
}



// no linter warnings:
let somethingElse = "";
somethingElse = "z";
console.log(somethingElse);

export const myFunction = (
  { thing }: { thing: string[] },
  somethingElse = false
) => {
  return somethingElse ? thing : ["stuff"];
};

// also no linter warnings:
export const myFunction = (
  x: { thing: string },
  somethingElse = false
) => {
  const {thing} = x;
  return somethingElse ? thing : ["stuff"];
};
```

**Comment by magurotuna** [comment]
Thanks for reporting.
This issue was already fixed in #596, #629 but it's not been adopted in the Deno CLI. 
Maybe in the next release of Deno it'll be available for you.

**Comment by ewired** [comment]
Ah okay, I should have noticed that, thanks.
